app/controllers/IndexController.scala

```diff
@@ -42,25 +42,33 @@ class IndexController @Inject() (
 
   def index(departureId: String): Action[AnyContent] = actions.getData(departureId).async {
     implicit request =>
-      service.getLRN(departureId) flatMap {
-        lrn =>
-          sessionRepository
-            .set(
-              request.userAnswers.getOrElse(
-                UserAnswers(
-                  departureId,
-                  request.eoriNumber,
-                  lrn.value,
-                  JsObject.empty,
-                  timeMachine.now()
+      service.getDepartureData(departureId).flatMap {
+        case Some(departureData) =>
+          service.getLRN(departureId) flatMap {
+            lrn =>
+              sessionRepository
+                .set(
+                  request.userAnswers.getOrElse(
+                    UserAnswers(
+                      departureId,
+                      request.eoriNumber,
+                      lrn.value,
+                      JsObject.empty,
+                      timeMachine.now()
+                    )
+                  )
                 )
-              )
-            )
-            .map(
-              _ => Redirect(controllers.routes.MoreInformationController.onPageLoad(departureId))
-            )
+                .map {
+                  _ =>
+                    if (departureData.data.isDataComplete) {
+                      Redirect(Call("GET", "#"))
+                    } else {
+                      Redirect(controllers.routes.MoreInformationController.onPageLoad(departureId))
+                    }
+                }
+          }
+        case None => ??? //technical difficulties
```

Comment:
Can we add this in quickly? Dont want to kick the can further down the road: 
https://github.com/hmrc/manage-transit-movements-departure-frontend/blob/main/app/controllers/ErrorController.scala